NineChronicles.Headless/GraphTypes/StandaloneQuery.cs

```diff
@@ -148,6 +153,57 @@ public StandaloneQuery(StandaloneContext standaloneContext, IConfiguration confi
                     return blockChain.GetTransaction(txId);
                 }
             );
+
+            Field<NonNullGraphType<ListGraphType<ShopItemType>>>(
+                name: "products",
+                arguments: new QueryArguments(
+                    new QueryArgument<IntGraphType>
+                    {
+                        Name = "id",
+                    },
+                    new QueryArgument<ItemSubTypeEnumType>
+                    {
+                        Name = "itemSubType"
+                    },
+                    new QueryArgument<IntGraphType>
+                    {
+                        Name = "price"
+                    }),
+                resolve: context =>
+                {
+                    if (!(standaloneContext.BlockChain is BlockChain<PolymorphicAction<ActionBase>> blockChain))
+                    {
+                        throw new ExecutionError(
+                            $"{nameof(StandaloneContext)}.{nameof(StandaloneContext.BlockChain)} was not set yet!");
+                    }
+
+                    var shop = new ShopState((Dictionary) blockChain.GetState(Addresses.Shop));
+                    var products = shop.Products.Values;
+                    var id = context.GetArgument<int>("id");
+                    if (id > 0)
+                    {
+                        products = products
+                            .Where(si => si.ItemUsable?.Id == id || si.Costume?.Id == id);
+                    }
+                    //Avoid default value for Consumable
+                    var subType = context.GetArgument<ItemSubType>("itemSubType", ItemSubType.EquipmentMaterial);
+                    if (subType != ItemSubType.EquipmentMaterial)
+                    {
+                        products = products
+                            .Where(si => si.ItemUsable?.ItemSubType == subType || si.Costume?.ItemSubType == subType);
+                    }
+                    var price = context.GetArgument<int>("price");
+                    if (price > 0)
+                    {
+                        var currency = new GoldCurrencyState(
+                            (Dictionary) blockChain.GetState(GoldCurrencyState.Address)
+                        ).Currency;
+                        products = products
+                            .Where(si => si.Price <= price * currency);
+                    }
```

Comment:
The `price` argument seems used as the maximum of prices. Then, how about renaming the argument by its purpose?